irohad/ametsuchi/impl/temporary_wsv_impl.cpp

```diff
@@ -59,15 +62,16 @@ namespace iroha {
       };
 
       transaction_->exec("SAVEPOINT savepoint_;");
-      auto result = apply_function(transaction, *wsv_)
-          && std::all_of(transaction.commands.begin(),
-                         transaction.commands.end(),
+      auto result = apply_function(tx, *wsv_)
+          && std::all_of(transaction->commands.begin(),
+                         transaction->commands.end(),
                          execute_command);
       if (result) {
         transaction_->exec("RELEASE SAVEPOINT savepoint_;");
       } else {
         transaction_->exec("ROLLBACK TO SAVEPOINT savepoint_;");
       }
+      delete transaction;
```

Comment:
`delete` triggered use smart pointers intead.

Reply:
Yeah, forgot to replace before commit